### ISSUE TITLE
Include macro

### ISSUE THREAD

**Issue by rkhamis** [threefoldtech/digitalmeX#35]
Include macro should be extended to allow users to define:
 - organization/repo
 - docsite
 - branch
 - document
 - document part

Please see spec [here](https://github.com/threefoldtech/jumpscaleX/blob/development_types/Jumpscale/tools/markdowndocs/macros/include.md)

**Comment by abom** [comment]
Macros tests [are fixed](https://github.com/threefoldtech/jumpscaleX/pull/218/commits/01938670b8b73092b14619bfc874d3b1ebb897d9), they can be used and updated now, also moved test files to tests directory.

**Comment by abom** [comment]
Newest specs can be found [here](https://github.com/threefoldtech/jumpscaleX/blob/development_types/docs/tools/wiki/wiki_include.md) and [here](https://github.com/threefoldtech/jumpscaleX/blob/development_types/docs/tools/wiki/code_wiki_parts.md).
Will update the docs in `tools/markdowndocs/macros` and re-link them again at the end.

**Comment by abom** [comment]
Should be done now on `development_types`, examples can be found at `docs/tools/wiki/docsites/examples`.
Another problem is the that macros are called 3 times instead of one! found the root cause, will try to fix without any side effects.

**Comment by abom** [comment]
> Another problem is the that macros are called 3 times instead of one! found the root cause, will try to fix without any side effects.

This should be fixed now too, processing is done only once.

**Comment by abom** [comment]
https://github.com/threefoldtech/jumpscaleX/pull/173

**Comment by abom** [comment]
This specs have been changed, closing this in favor of #42.
